accelerator/errors.h

```diff
@@ -8,14 +8,86 @@ extern "C" {
 /**
  * @file errors.h
  * @brief Error Code of tangle-acclerator
+ *
+ * bit division:
+ * 0 - 3 actual error code
+ * 4 - 5 serverity
+ * 6 - 8 error module
+ *
+ * *--------*
+ * |MMMSSCCC|
+ * *--------*
  */
 
-/** @name Http status code */
+/** @name serverity code */
 /** @{ */
-#define SC_BAD_REQUEST 400 /**< Http bad request */
-#define SC_NOT_FOUND 404   /**< Http not found */
+#define SC_SEVERITY_MASK 0x18
+#define SC_SEVERITY_SHIFT 3
+
+#define SC_SEVERITY_MINOR (0x0 << SC_SEVERITY_MASK)
+#define SC_SEVERITY_MODERATE (0x01 << SC_SEVERITY_MASK)
+#define SC_SEVERITY_MAJOR (0x02 << SC_SEVERITY_MASK)
+#define SC_SEVERITY_FATAL (0x03 << SC_SEVERITY_MASK)
+/** @} */
+
+/** @name module code */
+/** @{ */
+#define SC_MODULE_MASK 0xE0
+#define SC_MODULE_SHIFT 5
+
+#define SC_MODULE_TA (0x01 << SC_MODULE_SHIFT)
+#define SC_MODULE_CCLIENT (0x02 << SC_MODULE_SHIFT)
+#define SC_MODULE_SERIALIZER (0x03 << SC_MODULE_SHIFT)
+#define SC_MODULE_CACHE (0x04 << SC_MODULE_SHIFT)
+/** @} */
+
+/** @name serverity code */
+/** @{ */
+#define SC_ERROR_MASK 0x07
 /** @} */
 
+/** @name http error code */
+/** @{ */
+#define SC_BAD_REQUEST 400
+#define SC_NOT_FOUND 404
+/** @} */
+
+/* status code */
+typedef enum {
+  SC_OK = 0, /**< No error */
+
+  SC_TA_OOM = 0x01 | SC_MODULE_TA | SC_SEVERITY_FATAL,
```

Comment:
Let's utilize doxygen `@name` here and following modules below.

Reply:
It seems like doxygen does not support grouping in enum. :sob: